bigtable-hbase-parent/bigtable-hbase-integration-tests-common/src/main/java/com/google/cloud/bigtable/hbase/test_env/SharedTestEnvRule.java

```diff
@@ -34,31 +34,36 @@
 import org.apache.hadoop.hbase.util.Bytes;
 import org.junit.rules.ExternalResource;
 
-public abstract class SharedTestEnvRule extends ExternalResource {
+public class SharedTestEnvRule extends ExternalResource {
   private static final String HBASE_CONN_KEY = "hbase_conn";
 
   public static final int MAX_VERSIONS = 6;
   public static final byte[] COLUMN_FAMILY = Bytes.toBytes("test_family");
   public static final byte[] COLUMN_FAMILY2 = Bytes.toBytes("test_family2");
   private static SharedTestEnvRule instance;
 
+  private static TableCreator tableCreator = null;
+
+  static {
+    try {
+      Class<? extends TableCreator> clazz = (Class<? extends TableCreator>) Class
+          .forName("com.google.cloud.bigtable.hbase.test_env.TableCreatorImpl");
+      tableCreator = clazz.getDeclaredConstructor().newInstance();
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+  }
   /**
    * This class is generally a singleton, where implementation can change.  Some startup utility
    * will set this instance.
    */
-  public static void setInstance(SharedTestEnvRule instance) {
+  public synchronized static void setInstance(SharedTestEnvRule instance) {
```

Comment:
Can we remove this setInstance()? That way this class can be a better singleton with one synchronized call.